static-site/pages/groups/[...groupName].jsx

```diff
@@ -1,34 +1,37 @@
 import dynamic from 'next/dynamic'
 import { useRouter } from 'next/router'
 const IndividualGroupPage = dynamic(() => import("../../src/sections/individual-group-page"), {ssr: false})
+const GroupMembersPage = dynamic(() => import("../../src/sections/group-members-page.tsx"), {ssr: false})
 const GroupSettingsPage = dynamic(() => import("../../src/sections/group-settings-page"), {ssr: false})
 
 
 /**
  * NextJS dynamic routing doesn't allow us to set up routes where we have (e.g.)
  * groups/:groupName/settings -> <GroupSettingsPage>
+ * groups/:groupName/settings/members -> <GroupMembersPage>
  * groups/:groupName[/*] -> <IndividualGroupPage>
  * so we use some JS routing here to serve the right component
  */
 const Index = () => {
   const {query} = useRouter();
   if (!query.groupName) return null; // wait until query ready
 
-
-
-
   /* nextJs dynamic routing will set the part parts _beyond_ "groups/" as query.groupName */
-  if (query.groupName.length===2 && query.groupName[1]==='settings') {
-    // param location TODO!!!
-    return <GroupSettingsPage groupName={query.groupName[0]}/>
+  const groupName = query.groupName[0]
+  const resourcePath = query.groupName.slice(1).join("/")
+
+  if (resourcePath === 'settings') {
+    return <GroupSettingsPage groupName={groupName}/>
   }
 
-  const resourcePath = query.groupName.slice(1);
+  if (resourcePath === 'settings/members') {
+    return <GroupMembersPage groupName={groupName}/>
+  }
 
   return (
-    <IndividualGroupPage groupName={query.groupName[0]} resourcePath={resourcePath.length ? resourcePath : undefined}/>
+    <IndividualGroupPage groupName={groupName} resourcePath={resourcePath.length ? resourcePath : undefined}/>
   );
```

Comment:
I wonder if it's worth structuring this as either `if/else if/else` or a `switch` on `resourcePath`, to make it more clear that there are three mutually exclusive exits here.

Reply:
+1 for `switch`

Reply:
Good suggestion! Updated to `switch` in 195aa4f24c735222f8815bfaa2809eb26b2e5fd7